lib/web_ui/lib/src/engine/text/ruler.dart

```diff
@@ -852,7 +854,14 @@ class MeasurementResult {
   /// of each laid out line.
   final List<EngineLineMetrics> lines;
 
-  const MeasurementResult(
+  /// The text align value of the paragraph.
+  final ui.TextAlign textAlign;
+
+  /// The text direction of the paragraph.
+  final ui.TextDirection textDirection;
+
+  MeasurementResult.forParagraph(
+    EngineParagraph paragraph,
```

Comment:
Can we deconstruct the paragraph prior to calling the constructor? Otherwise, this API creates a temptation to store the paragraph in the `MeasurementResult` object, which could be very unsafe. At any moment the framework can draw the paragraph, lay it out again with different constraints, then draw it again, resulting in previous `MeasurementResult` objects holding onto a stale paragraph object.

Reply:
Some other APIs use the `from` nomenclature to indicate that the object is constructed based on another object but does not hold onto it (e.g. `List.from`), but to fully satisfy my paranoia I think I wouldn't use the paragraph at all :)

Reply:
Good point! I'll pass the `textDirection` and `textAlign` values individually.